src/Microsoft.AspNetCore.Mvc.RazorPages/Page.cs

```diff
@@ -232,6 +1009,71 @@ protected RedirectToPageResult RedirectToPagePermanent(string pageName, object r
             => new RedirectToPageResult(pageName, routeValues, permanent: true, fragment: fragment);
 
         /// <summary>
```

Comment:
`RedirectToPageResult` does not currently have a "preserve" property or constructor argument, so I assumed we didn't want a factory which considered it. If we should have that factory I'll add it and the property/ctor arg.